solidstart-tanstackquery-tailwind-modules/src/routes/[owner]/[name]/pulls.tsx

```diff
@@ -0,0 +1,120 @@
+import { Switch, Match, createSignal, createEffect } from 'solid-js';
+import { createQuery } from '@tanstack/solid-query';
+import { useParams, useSearchParams } from 'solid-start';
+import getRepoInfo from '~/services/get-repo-info';
+import { LoadingPulseDot } from '~/components/LoadingPulseDot/LoadingPulseDot';
+import { Info } from '~/types/repo-info-type';
+import { RepoHeader } from '~/components/RepoHeader';
+import getRepoPullRequests from '~/services/get-pull-request';
+import {
+  PullRequestsStore,
+  setPullRequestsStore,
+} from '~/stores/pull-requests-store';
+import { selectedLabel, sortBy } from '~/components/PRAndIssuesHeader';
+import { parseSortParams } from '~/components/RepoIssues/utils';
+import { DEFAULT_PAGE_SIZE, SORT_OPTIONS } from '~/utils/constants';
+import RepoPullRequests from '~/components/RepoPullRequests';
+
+const Issues = () => {
+  const params = useParams();
+  const [searchParams] = useSearchParams();
+  const [info, setInfo] = createSignal<Info>({
+    isPrivate: false,
+    visibility: '',
+    forkCount: 0,
+    description: '',
+    homepageUrl: '',
+    stargazerCount: 0,
+    watcherCount: 0,
+    openIssueCount: 0,
+    topics: [],
+    isOrg: false,
+    openPullRequestCount: 0,
+  });
+
+  const isOwnerAndNameValid =
+    typeof params.owner === 'string' && typeof params.name === 'string';
+
+  const repoInfo = createQuery(
+    () => [`repository-info_${params.owner}_${params.name}`],
+    () =>
+      getRepoInfo(
+        isOwnerAndNameValid
+          ? { owner: params.owner, name: params.name }
+          : { owner: '', name: '' }
+      )
+  );
+
+  const repoPullrequests = createQuery(
+    () => [
+      `repository-prs_${params.owner}_${params.name}`,
+      searchParams.after,
+      searchParams.before,
+      sortBy(),
+      selectedLabel(),
+    ],
+    () =>
+      getRepoPullRequests({
+        owner: params.owner,
+        name: params.name,
+        orderBy: parseSortParams(SORT_OPTIONS, sortBy(), 0),
+        direction: parseSortParams(SORT_OPTIONS, sortBy(), 1),
+        labels: selectedLabel() ? [selectedLabel()] : undefined,
+        before: searchParams.before,
+        after: searchParams.after,
+        first:
+          searchParams.after || !searchParams.before
+            ? DEFAULT_PAGE_SIZE
+            : undefined,
+        last: searchParams.before ? DEFAULT_PAGE_SIZE : undefined,
+      })
+  );
+
+  createEffect(() => {
+    if (repoInfo.isSuccess && !repoInfo.isLoading && repoInfo.data) {
+      setInfo(repoInfo.data.info);
+    }
+    if (
+      repoPullrequests.isSuccess &&
+      !repoPullrequests.isLoading &&
+      repoPullrequests.data
```

Comment:
Isn't [`createResource()`](https://www.solidjs.com/docs/latest/api#createresource) specifically for these kinds of async fetches?

Reply:
solid start has an issue, so the best way is to store the value in a state to use it. in SOlidJs, we didn't need to doo all this. And yes, Tanstack also handles all this
@BigAB